### PR TITLE
[#2041] Disable fields that are overridden by an Active Effect and display a tooltip explaining why they cannot be edited

### LINKED ISSUE #2041: Show a warning notification when attempting to change an overridden armour calculation, or have the field be disabled with a tooltip
If 'Unarmored Defense' (from both Monk and Barbarian) from the 'Class Features (SRD)' is added to the character sheet, it is no longer possible to change the AC calculation to anything else.
'Unarmored Defense' should only be active if no armor is being worn, however, equipping armor does not change the AC, and it's not possible to change the AC calculation to 'Equipped Armor'.
I believe that the issue is that the 'Change Mode' of the item's effect is set to 'Override'. I believe that it should either not be set to 'Override', or it should automatically switch to 'Equipped Armor' if the user has equipped any.

Tested on Foundry v10.291 and 5e v2.1.1 and v2.0.2

### PR DESCRIPTION
**Pull request by Fyorl**
- Closes #2041

### CODE REVIEW

**Review by aaclayton — APPROVED** [review]
Looks okay at surface level. If it's working in practice go ahead!

File: module/applications/actor/base-sheet.mjs [hunk 1]
```diff
@@ -643,6 +643,13 @@ export default class ActorSheet5e extends ActorSheet {
 
       // Active Effect management
       html.find(".effect-control").click(ev => ActiveEffect5e.onManageActiveEffect(ev, this.actor));
+
+      for ( const override of Object.keys(foundry.utils.flattenObject(this.actor.overrides)) ) {
+        html.find(`input[name="${override}"],select[name="${override}"]`).each((i, el) => {
+          el.disabled = true;
+          el.dataset.tooltip = "DND5E.ActiveEffectOverrideWarning";
```
Comment (arbron) [inline module/applications/actor/base-sheet.mjs:650]: For fields that have an existing tooltip, do we want to take that into account? Maybe prepending the tooltip with it?

**Review by arbron — COMMENTED** [review]
Seems good for most things, but doesn't work on the proficiency buttons on the character sheet because they aren't normal inputs.

File: module/applications/actor/base-sheet.mjs (hunk 1 above)
Comment (Fyorl) [reply module/applications/actor/base-sheet.mjs:650]: I tried a version where we didn't replace existing tooltips, and a version where we did (this version). I felt that replacing the tooltip worked best from a UX perspective. I didn't try a version where we concatenate the values at all, but it feels like that would be too long and also read strangely.

**Comment by Fyorl** [comment]
> Seems good for most things, but doesn't work on the proficiency buttons on the character sheet because they aren't normal inputs.

I will add some special-case handling for proficiency buttons.

File: module/applications/actor/base-sheet.mjs (hunk 1 above)
Comment (krbz999) [reply module/applications/actor/base-sheet.mjs:650]: Curious if this method covers the case where the input field isn't present, such as spell slot overrides, where the input field is created by clicking the 'edit' icon.

Comment (Fyorl) [reply module/applications/actor/base-sheet.mjs:650]: > Curious if this method covers the case where the input field isn't present, such as spell slot overrides, where the input field is created by clicking the 'edit' icon.

I will add some handling for this too 😩. Any others you can think of?

Comment (krbz999) [reply module/applications/actor/base-sheet.mjs:650]: Anything in Special Traits and other sub-applications?

Comment (Fyorl) [reply module/applications/actor/base-sheet.mjs:650]: Those are already handled with the current version.

**Review by arbron — APPROVED** [review]

**Pull request merged**